### PR TITLE
optimization debian package manager tweaks

### PR DESCRIPTION
**Pull request by rajpratik71**
By default, Ubuntu or Debian based "apt" or "apt-get" system installs recommended but not suggested packages .

By passing "--no-install-recommends" option, the user lets apt-get know not to consider recommended packages as a dependency to install.

This results in smaller downloads and installation of packages .

Refer to blog at [Ubuntu Blog](https://ubuntu.com/blog/we-reduced-our-docker-images-by-60-with-no-install-recommends) .

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by oddlittlebird** [comment]
@xlson , please review.

**Review by xlson — COMMENTED** [review]
Thanks for contribution. I'll take a look at this when I get some time over, but it's not very high on the priority list as it will take quite some time to verify that we don't depend on one of the recommended packages.

@Rajpratik71 if you want to make sure this gets reviewed and merged sooner it would be a good idea to break this apart into a few different PRs. One for the main Dockerfiles (`Dockerfile`, `packaging/docker/*`) that we use to build grafana, one for the docs and one for the ci docker containers (`scripts/build/ci-*`).

**Comment by aknuds1** [comment]
Please break this into several PRs, since it touches on too many different files (both Dockerfiles for internally used images and externally used images, and documentation).

**Comment by Rajpratik71** [comment]
@aknuds1 requesting review

**Comment by Rajpratik71** [comment]
for the failing build issue , i looked at logs , it is related to "TypeScript" which is not touched means unrelated failing builds

**Comment by oddlittlebird** [comment]
Try updating the branch from master.

**Comment by Rajpratik71** [comment]
> Try updating the branch from master.

Updated to upstream . Please review . Is pending from months

**Comment by Rajpratik71** [comment]
@xlson @oddlittlebird @aknuds1  requesting review

File: .circleci/config.yml [hunk 1]
```diff
@@ -735,25 +735,25 @@ jobs:
...
-          command: "./scripts/circle-test-mysql.sh"
-      - run:
-          name: ci job failed
-          command: "./scripts/ci-job-failed.sh"
-          when: on_fail
-      - run:
-          name: ci job succeeded
-          command: "./scripts/ci-job-succeeded.sh"
-          when: on_success
+        - checkout
```
Comment (aknuds1) [inline .circleci/config.yml:738]: Please revert your formatting changes.

File: Dockerfile.ubuntu [hunk 2]
```diff
@@ -49,8 +49,8 @@ WORKDIR $GF_PATHS_HOME
 
 COPY conf conf
 
-# curl should be part of the image
-RUN apt-get update && apt-get upgrade -y && apt-get install -y ca-certificates curl
+# We need font libs for phantomjs, and curl should be part of the image
```
Comment (aknuds1) [inline Dockerfile.ubuntu:52]: Please revert this change.

File: scripts/build/ci-build/Dockerfile [hunk 3]
```diff
@@ -56,8 +56,8 @@ RUN UNATTENDED=yes OSX_VERSION_MIN=${OSX_MIN} /tmp/osxcross/build.sh
 RUN rm -rf /tmp/osxcross/target/SDK/${OSX_SDK}/usr/share && \
     cd /tmp                                              && \
     tar cfJ osxcross.tar.xz osxcross/target              && \
-    rm -rf /tmp/osxcross
-RUN apt-get install -y                     \
+    rm -rf /tmp/osxcross && \
```
Comment (aknuds1) [inline scripts/build/ci-build/Dockerfile:59]: Please revert this change.

**Review by aknuds1 — CHANGES_REQUESTED** [review]

**Comment by aknuds1** [comment]
Just a cursory review for now. In order to review in depth, I need to find the time, since I have to verify that your changes don't break things.

File: .circleci/config.yml (hunk 1 above)
Comment (Rajpratik71) [reply .circleci/config.yml]: fixed

File: Dockerfile.ubuntu (hunk 2 above)
Comment (Rajpratik71) [reply Dockerfile.ubuntu]: fixed

File: scripts/build/ci-build/Dockerfile (hunk 3 above)
Comment (Rajpratik71) [reply scripts/build/ci-build/Dockerfile]: fixed

**Comment by Rajpratik71** [comment]
> for the failing build issue , i looked at logs , it is related to "TypeScript" which is not touched means unrelated failing builds

Again this happened after rebasing from Upstream

**Comment by Rajpratik71** [comment]
@xlson @aknuds1 @oddlittlebird  requesting review

File: .circleci/config.yml [hunk 4]
```diff
@@ -728,7 +728,7 @@ jobs:
           name: ci job started
           command: "./scripts/ci-job-started.sh"
       - run: sudo apt update
-      - run: sudo apt install -y default-mysql-client
+      - run: sudo apt-get --no-install-recommends install -y default-mysql-client
```
Comment (aknuds1) [inline .circleci/config.yml:731]: Please remove this change.

File: .circleci/config.yml [hunk 5]
```diff
@@ -757,7 +757,7 @@ jobs:
           name: ci job started
           command: "./scripts/ci-job-started.sh"
       - run: sudo apt update
-      - run: sudo apt install -y postgresql-client
+      - run: sudo apt-get --no-install-recommends install -y postgresql-client
```
Comment (aknuds1) [inline .circleci/config.yml:760]: Please remove this change.

File: Dockerfile.ubuntu [hunk 6]
```diff
@@ -50,7 +50,7 @@ WORKDIR $GF_PATHS_HOME
 COPY conf conf
 
 # curl should be part of the image
-RUN apt-get update && apt-get install -y ca-certificates curl
+RUN apt-get update && apt-get upgrade -y && apt-get --no-install-recommends install -y ca-certificates curl
```
Comment (aknuds1) [inline Dockerfile.ubuntu:53]:
```suggestion
RUN apt-get update && apt-get upgrade -y && apt-get install --no-install-recommends -y ca-certificates curl
```

File: devenv/docker/blocks/multiple-openldap/admins-ldap-server.Dockerfile [hunk 7]
```diff
@@ -7,7 +7,7 @@ LABEL maintainer="Grafana team <hello@grafana.com>"
 ENV OPENLDAP_VERSION 2.4.40
 
 RUN apt-get update && \
-    DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y \
+    DEBIAN_FRONTEND=noninteractive apt-get --no-install-recommends install -y \
```
Comment (aknuds1) [inline devenv/docker/blocks/multiple-openldap/admins-ldap-server.Dockerfile:10]: Please remove this change.

File: devenv/docker/blocks/multiple-openldap/ldap-server.Dockerfile [hunk 8]
```diff
@@ -7,7 +7,7 @@ LABEL maintainer="Grafana team <hello@grafana.com>"
 ENV OPENLDAP_VERSION 2.4.40
 
 RUN apt-get update && \
-    DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y \
+    DEBIAN_FRONTEND=noninteractive apt-get --no-install-recommends install -y \
```
Comment (aknuds1) [inline devenv/docker/blocks/multiple-openldap/ldap-server.Dockerfile:10]: Please remove this change.

File: devenv/docker/blocks/mysql_opendata/Dockerfile [hunk 9]
```diff
@@ -8,7 +8,7 @@ ENV MYSQL_DATABASE="testdata" \
     MYSQL_PASSWORD="password"
 
 # Install requirement (wget)
-RUN apt-get update && apt-get install -y wget && apt-get install unzip
+RUN apt-get update && apt-get --no-install-recommends install -y wget && apt-get --no-install-recommends install -y unzip
```
Comment (aknuds1) [inline devenv/docker/blocks/mysql_opendata/Dockerfile:11]: Please remove this change.

File: devenv/docker/blocks/openldap/Dockerfile [hunk 10]
```diff
@@ -7,7 +7,7 @@ LABEL maintainer="Grafana team <hello@grafana.com>"
 ENV OPENLDAP_VERSION 2.4.40
 
 RUN apt-get update && \
-    DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y \
+    DEBIAN_FRONTEND=noninteractive apt-get --no-install-recommends install -y \
```
Comment (aknuds1) [inline devenv/docker/blocks/openldap/Dockerfile:10]: Please remove this change.

File: devenv/docker/debtest/Dockerfile [hunk 11]
```diff
@@ -1,6 +1,6 @@
 FROM debian:jessie
 
-RUN apt-get update && apt-get install -y vim
+RUN apt-get update && apt-get --no-install-recommends install -y vim
```
Comment (aknuds1) [inline devenv/docker/debtest/Dockerfile:3]: Please remove this change.

File: devenv/docker/ha_test/README.md [hunk 12]
```diff
@@ -26,7 +26,7 @@ $ make build-docker-full
 #### Alternative 1 - Use dnsmasq
 
 ```bash
-$ sudo apt-get install dnsmasq
+$ sudo apt-get --no-install-recommends install -y dnsmasq
```
Comment (aknuds1) [inline devenv/docker/ha_test/README.md:29]: Please remove this change.

File: docs/sources/installation/debian.md [hunk 13]
```diff
@@ -36,8 +36,8 @@ If you install from the APT repository, then Grafana is automatically updated ev
 #### To install the latest Enterprise edition:
 
 ```bash
-sudo apt-get install -y apt-transport-https
-sudo apt-get install -y software-properties-common wget
+sudo apt-get --no-install-recommends install -y apt-transport-https
+sudo apt-get --no-install-recommends install -y software-properties-common wget
```
Comment (aknuds1) [inline docs/sources/installation/debian.md:40]: Please remove these changes.

File: docs/sources/installation/debian.md [hunk 14]
```diff
@@ -56,14 +56,14 @@ After you add the repository:
...
+sudo apt-get --no-install-recommends install -y grafana-enterprise
 ```
 
 #### To install the latest OSS release:
 
 ```bash
-sudo apt-get install -y apt-transport-https
-sudo apt-get install -y software-properties-common wget
+sudo apt-get --no-install-recommends install -y apt-transport-https
+sudo apt-get --no-install-recommends install -y software-properties-common wget
```
Comment (aknuds1) [inline docs/sources/installation/debian.md:66]: Please remove these changes.

File: docs/sources/installation/debian.md [hunk 15]
```diff
@@ -82,7 +82,7 @@ After you add the repository:
 
 ```bash
 sudo apt-get update
-sudo apt-get install grafana
+sudo apt-get --no-install-recommends install grafana
```
Comment (aknuds1) [inline docs/sources/installation/debian.md:85]: Please remove this change.

File: docs/sources/installation/debian.md [hunk 16]
```diff
@@ -99,7 +99,7 @@ If you install the `.deb` package, then you will need to manually update Grafana
 1. Copy and paste the code from the installation page into your command line and run. It follows the pattern shown below.
 
 ```bash
-sudo apt-get install -y adduser libfontconfig1
+sudo apt-get --no-install-recommends install -y adduser libfontconfig1
```
Comment (aknuds1) [inline docs/sources/installation/debian.md:102]: Please remove this change.

File: docs/sources/installation/upgrading.md [hunk 17]
```diff
@@ -65,7 +65,7 @@ links.
 
 ```bash
 wget <debian package url>
-sudo apt-get install -y adduser libfontconfig1
+sudo apt-get --no-install-recommends install -y adduser libfontconfig1
```
Comment (aknuds1) [inline docs/sources/installation/upgrading.md:68]: Please remove this change.

File: docs/sources/tutorials/stack_guide_graphite.md [hunk 18]
```diff
@@ -39,7 +39,7 @@ queries that will make the most out of your collected metrics.
 Graphite and Carbon are written in python, so we will start by installing python packages.
 
 ```
-apt-get install \
+apt-get --no-install-recommends install -y \
```
Comment (aknuds1) [inline docs/sources/tutorials/stack_guide_graphite.md:42]: Please remove this change.

File: docs/sources/tutorials/stack_guide_graphite.md [hunk 19]
```diff
@@ -146,7 +146,7 @@ end with max, min or count then use average. This means that naming metrics is v
 ### Start carbon
 Lets install supervisord and let it start carbon.
 
-`apt-get install supervisor`
+`apt-get --no-install-recommends install -y supervisor`
```
Comment (aknuds1) [inline docs/sources/tutorials/stack_guide_graphite.md:149]: Please remove this change.

File: packaging/docker/ubuntu.Dockerfile [hunk 20]
```diff
@@ -28,7 +28,7 @@ WORKDIR $GF_PATHS_HOME
 
 # Install dependencies
 # We need curl in the image
-RUN apt-get update && apt-get install -y ca-certificates curl tzdata && \
+RUN apt-get update && apt-get --no-install-recommends install -y ca-certificates curl tzdata && \
```
Comment (aknuds1) [inline packaging/docker/ubuntu.Dockerfile:31]:
```suggestion
RUN apt-get update && apt-get install --no-install-recommends  -y ca-certificates curl tzdata && \
```

File: scripts/build/ci-build/Dockerfile [hunk 21]
```diff
@@ -35,7 +35,7 @@ WORKDIR /tmp
 # might also wanna put them as their own layer to not have to unpack them every time?
 
 RUN apt-get update && \
-    apt-get install -yq \
+    apt-get --no-install-recommends install -yq \
```
Comment (aknuds1) [inline scripts/build/ci-build/Dockerfile:38]: Please remove this change.

File: scripts/build/ci-build/Dockerfile [hunk 22]
```diff
@@ -54,7 +54,8 @@ RUN rm -rf /tmp/osxcross/target/SDK/${OSX_SDK}/usr/share && \
     cd /tmp                                              && \
     tar cfJ osxcross.tar.xz osxcross/target              && \
     rm -rf /tmp/osxcross
-RUN apt-get install -yq                     \
+
+RUN apt-get --no-install-recommends install -yq                     \
```
Comment (aknuds1) [inline scripts/build/ci-build/Dockerfile:58]: Please remove this change.

File: scripts/build/ci-build/Dockerfile [hunk 23]
```diff
@@ -99,7 +100,7 @@ COPY --from=toolchain /tmp/x86_64-centos6-linux-gnu.tar.xz /tmp/osxcross.tar.xz
 COPY --from=toolchain /tmp/golangci-lint /usr/local/bin
 
 RUN apt-get update && \
-    apt-get install -yq  \
+    apt-get --no-install-recommends install -yq  \
```
Comment (aknuds1) [inline scripts/build/ci-build/Dockerfile:103]: Please remove this change.

File: scripts/build/ci-deploy/Dockerfile [hunk 24]
```diff
@@ -14,7 +14,7 @@ RUN pip install -U awscli crcmod && \
     curl https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-222.0.0-linux-x86_64.tar.gz | \
       tar xvzf - -C /opt && \
     apt update && \
-    apt install -y createrepo expect && \
+    apt-get --no-install-recommends install -y createrepo expect && \
```
Comment (aknuds1) [inline scripts/build/ci-deploy/Dockerfile:17]: Please remove this change.

File: scripts/verify-repo-update/Dockerfile.deb [hunk 25]
```diff
@@ -4,12 +4,12 @@ ARG REPO_CONFIG=grafana.list.oss
 ARG PACKAGE=grafana
 
 RUN apt update && \
-    apt install -y curl                \
+    apt-get --no-install-recommends install -y curl                \
```
Comment (aknuds1) [inline scripts/verify-repo-update/Dockerfile.deb:7]: Please remove this change.

File: scripts/verify-repo-update/Dockerfile.deb [hunk 26]
```diff
@@ -4,12 +4,12 @@ ARG REPO_CONFIG=grafana.list.oss
...
                    apt-transport-https \
                    ca-certificates     \
                    gnupg               && \
     curl https://packages.grafana.com/gpg.key | apt-key add -
 
 COPY "./$REPO_CONFIG" /etc/apt/sources.list.d/grafana.list
 RUN apt update && \
-    apt install -y $PACKAGE
+    apt-get --no-install-recommends install -y $PACKAGE
```
Comment (aknuds1) [inline scripts/verify-repo-update/Dockerfile.deb:15]: Please remove this change.

**Review by aknuds1 — CHANGES_REQUESTED** [review]
We've decided that most changes from the PR should be dropped.

**Comment by Rajpratik71** [comment]
@aknuds1 removed everyone which is not needed or was duplicate. Only left with where needed.

**Comment by aknuds1** [comment]
Why have you resolved my review comments? Please do as I ask in the comments, or I will have to close the PR.

File: devenv/docker/blocks/mysql_opendata/Dockerfile [hunk 27]
```diff
@@ -8,7 +8,7 @@ ENV MYSQL_DATABASE="testdata" \
     MYSQL_PASSWORD="password"
 
 # Install requirement (wget)
-RUN apt-get update && apt-get install -y wget && apt-get install unzip
+RUN apt-get update && apt-get --no-install-recommends install -y wget unzip
```
Comment (aknuds1) [inline devenv/docker/blocks/mysql_opendata/Dockerfile:11]: Please remove this change.

**Review by aknuds1 — CHANGES_REQUESTED** [review]

**Review by aknuds1 — CHANGES_REQUESTED** [review]
Please do as I ask in review comments.

File: devenv/docker/blocks/mysql_opendata/Dockerfile (hunk 27 above)
Comment (Rajpratik71) [reply devenv/docker/blocks/mysql_opendata/Dockerfile]: done as requested

**Comment by Rajpratik71** [comment]
removing everything leave nothing in commit , closing it .

**Pull request closed without merging**